ui/job-view/headerbars/SecondaryNavBar.jsx

```diff
@@ -159,16 +159,52 @@ export default class SecondaryNavBar extends React.Component {
   }
 
   unwatchRepo(name) {
-    this.ThRepositoryModel.unwatchRepo(name);
-    this.setState({ watchedRepos: this.ThRepositoryModel.watchedRepos });
+    const { watchedRepoNames } = this.state;
+
+    this.setState({ watchedRepoNames: watchedRepoNames.filter(repo => repo.name !== name) });
```

Comment:
In the filter, does `repo` have a `name` key or should it be `repo[name]`?

Reply:
Always good to have a double-check on stuff like that though.  :)